modular_citadel/code/modules/reagents/reagent container/hypospraymkii.dm

```diff
@@ -109,10 +109,23 @@
 		return FALSE
 	return FALSE
 
-/obj/item/reagent_containers/hypospray/mkii/attack(obj/item/I, mob/user, params)
+/obj/item/hypospray/mkii/AltClick(mob/user)
+	if(vial)
+		vial.attack_self(user)
+
+/obj/item/hypospray/mkii/emag_act(mob/user)
```

Comment:
Really don't need more functions hidden behind an emag for reasons I've set out in my reply

Reply:
Quickreload is no longer there, instead, piercing is. Could make EMPs also change the piercing mode but I'm not going to consider it for now.